src/envoy/core.clj

```diff
@@ -37,6 +38,26 @@
            [(if to-keys? (keyword Key) Key)
             (when Value (fromBase64 Value))]))))
 
+(defn- round-robin
+  [hosts state]
+  (if (= 0 (mod @state (count hosts)))
+    (do (reset! state 0)
+      (get hosts 0))
+    (do (swap! state inc)
+      (get hosts (- @state 1)))))
+
+(defn client
```

Comment:
looks like this returns URL + functions are usually verbs, I would name it `find-consul-url`